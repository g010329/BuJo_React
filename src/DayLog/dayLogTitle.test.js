import React from "react";
import LogTitle from "./dayLogTitle.js";
import renderer from 'react-test-renderer';

test("Test LogTitle Component", ()=>{
    let today = new Date();
    let tree = renderer.create(
        <LogTitle data={
            {state:{year:today.getFullYear(),month:today.getMonth(),date:today.getDate(),theme:'dk'},
            method:{
                ifChangeDate:()=>{},
                handleDateBackward:()=>{},
                handleDateForward:()=>{},
                toggleIfInput:()=>{}
            }}}
        />).toJSON();
    expect(tree).toMatchSnapshot();

    tree = renderer.create(
        <LogTitle data={
            {state:{year:2020,month:8,date:1,theme:'dk'},
            method:{
                ifChangeDate:()=>{},
                handleDateBackward:()=>{},
                handleDateForward:()=>{},
                toggleIfInput:()=>{}
            }}}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});