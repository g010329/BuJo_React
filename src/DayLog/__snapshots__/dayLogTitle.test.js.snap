// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LogTitle Component 1`] = `
<div
  className="month_title month_title_dk"
>
  <div>
    <span
      className="onlyShowToday"
    >
      Today 
    </span>
    <span
      className="title_month"
    >
      8
      /
      10
    </span>
    
  </div>
  <span
    className="title_right"
  >
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-calendar popUp"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-angle-left"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-angle-right"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-plus"
        onClick={[Function]}
      />
    </span>
  </span>
</div>
`;

exports[`Test LogTitle Component 2`] = `
<div
  className="month_title month_title_dk"
>
  <div>
    
    <span
      className="title_month"
    >
      9
      /
      1
    </span>
    <span
      className="showYear"
    >
      2020
    </span>
  </div>
  <span
    className="title_right"
  >
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-calendar popUp"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-angle-left"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-angle-right"
        onClick={[Function]}
      />
    </span>
    <span
      className="icon_hover_span icon_hover_span_dk"
    >
      <i
        className="fas fa-plus"
        onClick={[Function]}
      />
    </span>
  </span>
</div>
`;
